drivers/display/display_st7701.c

```diff
@@ -0,0 +1,459 @@
+/*
+ * Copyright (c) 2025 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sitronix_st7701
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/mipi_dsi.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(st7701, CONFIG_DISPLAY_LOG_LEVEL);
+
+#include "display_st7701.h"
+
+struct st7701_config {
+	const struct device *mipi_dsi;
+	const struct gpio_dt_spec reset;
+	const struct gpio_dt_spec backlight;
+	uint8_t data_lanes;
+	uint16_t width;
+	uint16_t height;
+	uint8_t channel;
+	uint16_t rotation;
+};
+
+struct st7701_data {
+	uint16_t xres;
+	uint16_t yres;
+	uint8_t dsi_pixel_format;
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+};
+
+static inline int st7701_dcs_write(const struct device *dev, uint8_t cmd, const void *buf,
+		size_t len)
```

Comment:
Fix alignment

```suggestion
static inline int st7701_dcs_write(const struct device *dev, uint8_t cmd, const void *buf,
				   size_t len)
```

Reply:
These files were formatted by `clang-format`, using your format config `.clang-format`. I can fix it manually but I checked other drivers and they don't use this style.

Reply:
I did run the formatter, twice:

```bash
$ clang-format -i drivers/display/display_st7701.*
$ git diff drivers/display/display_st7701.*
```
Prints nothing...  Am I missing something? Or could it be a version issue? I assume it's using the `.clang_format` file in the root dir.

But clang aside, this seems to be how it should be formatted, for example: https://github.com/zephyrproject-rtos/zephyr/blob/44dfa8b2f4ccd9747c2a8e2f863f8245892ebaa5/drivers/display/display_renesas_ra.c#L56

Reply:
> I did run the formatter, twice:
> 
> ```shell
> $ clang-format -i drivers/display/display_st7701.*
> $ git diff drivers/display/display_st7701.*
> ```
> 
> Prints nothing... Am I missing something? Or could it be a version issue? I assume it's using the `.clang_format` file in the root dir.

With the current version, indeed. The version before the suggestion does yield a difference. 



> But clang aside, this seems to be how it should be formatted, for example:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/44dfa8b2f4ccd9747c2a8e2f863f8245892ebaa5/drivers/display/display_renesas_ra.c#L56

Functions parameters need to be aligned after the opening `(` bracket, using tabs and trailing spaces.

Reply:
Odd 🤷